pkg/controller/releasemanager/plan/syncApp.go

```diff
@@ -570,3 +577,24 @@ func (p *SyncApp) prometheusRule() *monitoringv1.PrometheusRule {
 		},
 	}
 }
+
+func (p *SyncApp) sidecarResource() *istioclient.Sidecar {
+	if p.IstioSidecarConfig == nil {
+		return nil
+	}
+
+	return &istioclient.Sidecar{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.App,
+			Namespace: p.Namespace,
+			Labels:    p.Labels,
+		},
+		Spec: istio.Sidecar{
+			Egress: []*istio.IstioEgressListener{
```

Comment:
Does this send all egress through the egress gateways automatically!?

Reply:
I wish. This only limits the xDS updates to Istio sidecars inside the namespace with the Sidecar resource.